src/ResidueField.jl

```diff
@@ -490,6 +490,7 @@ residue ring parent object is cached and returned for any subsequent calls
 to the constructor with the same base ring $R$ and element $a$.
 """
 function residue_field(R::Ring, a::RingElement; cached::Bool = true)
+   @req !is_trivial(R) "Zero rings are currently not supported as base ring."
```

Comment:
I think here, too, we could remove the "currently", as any quotient of a zero ring can't be a field.